regression-tests/run-tests.sh

```diff
@@ -0,0 +1,306 @@
+fi
+
+################
+cppfront_cmd="cppfront.exe"
+echo "Building cppfront"
+$compiler_cmd"$cppfront_cmd" ../source/cppfront.cpp
+if [[ $? -ne 0 ]]; then
+    echo "Compilation failed"
+    exit 2
+fi
+
+################
+failed_tests=()
+failed_compilations=()
+skipped_tests=()
+echo "Running regression tests"
+for test_file in $tests; do
+    test_name=${test_file%.*}
+    expeced_output="$expected_results_dir/$test_file.output"
+    generated_cpp_name=$test_name.cpp
+    expected_src="$expected_results_dir/$generated_cpp_name"
+    test_bin="test.exe"
+
+    # Choose mode - default to mixed code
+    descr="mixed cpp1 and cpp2 code"
+    opt=""
+    # Using naming convention to discriminate pure cpp2 code
+    if [[ $test_name == "pure2"* ]]; then
+        descr="pure cpp2 code"
+        opt="-p"
+    fi
+    echo "    Testing $descr: $test_name.cpp2"
+
+    ########
+    # Run the translation test
+    echo "        Generating C++1 code"
+    ./"$cppfront_cmd" "$test_file" -o "$expected_src" $opt > "$expeced_output" 2>&1
+
+    failure=0
+    compiler_issue=0
+    ########
+    # Check the output
+    if [ -f "$expeced_output" ]; then
+        # The C++1 generation output has to be tracked
+        check_file "$expeced_output"
+        if [[ $untracked -eq 1 ]]; then
+            echo "            The C++1 generation output file is not tracked by git:"
+            echo "                $expeced_output"
+            failure=1
+        elif [[ -n "$diff_output" ]]; then
+                echo "            Non-matching std out/err:"
+                printf "\n$diff_output\n\n"
+                failure=1
+        fi
+    else
+        echo "            Missing expected output file treaded as failure"
+        failure=1
+    fi
+
+    ########
+    # Check the generated code
+    if [ -f "$expected_src" ]; then
+        # The file was generated, so it shoudl be tracked by git
+        check_file "$expected_src"
+        if [[ $untracked -eq 1 ]]; then
+            echo "            The generated file is not tracked by git:"
+            echo "                $expected_src"
+            failure=1
+        elif [[ -n "$diff_output" ]]; then
+            echo "            Non-matching generated src:"
+            printf "\n$diff_output\n\n"
+            failure=1
+        else
+            ########
+            # Compile and run the generated code in a sub-shell
+            expected_src_compil_out="$exec_out_dir/$generated_cpp_name.output"
+            expected_src_exec_out="$exec_out_dir/$generated_cpp_name.execution"
+            expected_files="$expected_results_dir/$test_name.files"
+
+            echo "        Compiling generated code"
+            
+            # For some tests the binary needs to be placed in "$exec_out_dir"
+            # For that reason the compilation is done directly in that dir
+            # The source is temporarily copied to avoid issues with bash paths in cl.exe
+            (cd $exec_out_dir; \
```

Comment:
This compilation command could be greatly simplified after a modification discussed in [this comment](https://github.com/hsutter/cppfront/pull/897/files#r1432705805).